packages/react/src/components/Storage/FileUploader/Tracker/index.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback } from 'react';
+import { translate } from '@aws-amplify/ui';
+import { humanFileSize } from '@aws-amplify/ui';
+import { TrackerProps } from '../types';
+import {
+  View,
+  Image,
+  Text,
+  Button,
+  TextField,
+  Loader,
+  ComponentClassNames,
+  VisuallyHidden,
+} from '../../../../primitives';
+import {
+  IconClose,
+  IconEdit,
+  IconFile,
+} from '../../../../primitives/Icon/internal';
+import { FileMessage } from './FileMessage';
+
+export function Tracker({
+  errorMessage,
+  file,
+  fileState,
+  hasImage,
+  name,
+  onCancel,
+  onCancelEdit,
+  onPause,
+  onResume,
+  onSaveEdit,
+  onStartEdit,
+  percentage,
+  isResumable,
+  showImage,
+}: TrackerProps): JSX.Element {
+  const [tempName, setTempName] = React.useState(name);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const url = URL.createObjectURL(file);
+
+  // Focus the input after pressing the edit button
+  React.useEffect(() => {
+    if (fileState === 'editing' && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [fileState]);
+
+  const { size } = file;
+
+  const icon = hasImage ? <Image alt={file.name} src={url} /> : <IconFile />;
+  const isDeterminate = !isResumable || (isResumable && percentage > 0);
```

Comment:
Think i know why the LGTM bot keeps flagging this line, if the first condition resolves to `false` (`!isResumable`), then there is no need to check for `resumable`

Reply:
I'll try a tertiary...

Reply:
What abt:
```
 const isDeterminate = !isResumable || percentage > 0;
```

Reply:
[885920e](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/885920e203a8c6326d6086725c05a11f0b2e602f)